### PR TITLE
Add flag to IP Pools to allow pool to be excluded from Calico IPAM assignment

### PR DESCRIPTION
**Pull request by robbrockbank**

### CODE REVIEW

File: calico_containers/pycalico/datastore_datatypes.py [hunk 1]
```diff
@@ -156,14 +161,16 @@ def from_json(cls, json_str):
         json_dict = json.loads(json_str)
         return cls(json_dict["cidr"],
                    ipip=json_dict.get("ipip"),
-                   masquerade=json_dict.get("masquerade"))
+                   masquerade=json_dict.get("masquerade", False),
```
Comment (tomdee) [inline calico_containers/pycalico/datastore_datatypes.py]: Why this change to add the default?

File: calico_containers/pycalico/datastore.py [hunk 2]
```diff
@@ -326,11 +326,13 @@ def get_host_bgp_ips(self, hostname):
             return (ipv4, ipv6)
 
     @handle_errors
-    def get_ip_pools(self, version):
+    def get_ip_pools(self, version, ipam_only=False):
```
Comment (tomdee) [inline calico_containers/pycalico/datastore.py]: Why have ipam only in one place and skip ipam in another? Would it be better to be consistent?

**Comment by tomdee** [comment]
Couple of minor comments but otherwise LGTM

File: calico_containers/pycalico/datastore_datatypes.py (hunk 1 above)
Comment (robbrockbank) [inline calico_containers/pycalico/datastore_datatypes.py]: No reason whatsoever!  I'll remove.

File: calico_containers/pycalico/datastore.py (hunk 2 above)
Comment (robbrockbank) [inline calico_containers/pycalico/datastore.py]: Yeah I will change that!  It was originally because I wanted a lack of value in the JSON to be "False", but that really isn't necessary.
